core/io/image.cpp

```diff
@@ -2515,19 +2544,21 @@ Error Image::decompress() {
 		_image_decompress_etc1(this);
 	} else if (format >= FORMAT_ETC2_R11 && format <= FORMAT_ETC2_RA_AS_RG && _image_decompress_etc2) {
 		_image_decompress_etc2(this);
+	} else if (format >= FORMAT_ASTC_4x4 && format <= FORMAT_MAX && _image_decompress_astc) {
```

Comment:
Do we really want to decompress `FORMAT_MAX` as ASTC? :)
```suggestion
	} else if (format >= FORMAT_ASTC_4x4 && format <= FORMAT_ASTC_8x8_HDR && _image_decompress_astc) {
```
(The intention was likely to use `<`, but I wouldn't presume that ASTC is the last one before `MAX` as this could change in the future.)